diff --git a/node_modules/@xenova/transformers/src/backends/onnx.js b/node_modules/@xenova/transformers/src/backends/onnx.js
index 92cbfb2..8e05e5c 100644
--- a/node_modules/@xenova/transformers/src/backends/onnx.js
+++ b/node_modules/@xenova/transformers/src/backends/onnx.js
@@ -17,16 +17,31 @@
  */
 
 // NOTE: Import order matters here. We need to import `onnxruntime-node` before `onnxruntime-web`.
-export * as ONNX from 'onnxruntime-common';
+export * as ONNX from '@fugood/onnxruntime-common';
 
 export const executionProviders = [
     // 'webgpu',
-    'wasm'
+    'wasm',
 ];
 
 if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
-    import('onnxruntime-react-native');
+    import('@fugood/onnxruntime-react-native');
     executionProviders.unshift('cpu');
+    executionProviders.splice(executionProviders.indexOf('wasm'), 1);
+
+    import('react-native').then(({ Platform }) => {
+        if (Platform.OS === 'ios') {
+            executionProviders.unshift('coreml', 'xnnpack');
+        } else if (Platform.OS === 'android') {
+            executionProviders.unshift(
+                {
+                    name: 'nnapi',
+                    useFP16: true,
+                },
+                'xnnpack',
+            );
+        }
+    });
 
 } else if (typeof process !== 'undefined' && process?.release?.name === 'node') {
     // Running in a node-like environment.
