diff --git a/node_modules/react-native-sound-player/android/src/main/java/com/johnsonsu/rnsoundplayer/RNSoundPlayerModule.java b/node_modules/react-native-sound-player/android/src/main/java/com/johnsonsu/rnsoundplayer/RNSoundPlayerModule.java
index 2a1561c..dd3834e 100644
--- a/node_modules/react-native-sound-player/android/src/main/java/com/johnsonsu/rnsoundplayer/RNSoundPlayerModule.java
+++ b/node_modules/react-native-sound-player/android/src/main/java/com/johnsonsu/rnsoundplayer/RNSoundPlayerModule.java
@@ -164,6 +164,8 @@ public class RNSoundPlayerModule extends ReactContextBaseJavaModule implements L
         this.mediaPlayer = MediaPlayer.create(getCurrentActivity(), this.getUriFromFile(name, type));
       }
 
+      if (this.mediaPlayer == null) return
+
       this.mediaPlayer.setOnCompletionListener(
         new OnCompletionListener() {
           @Override
@@ -217,6 +219,7 @@ public class RNSoundPlayerModule extends ReactContextBaseJavaModule implements L
     if (this.mediaPlayer == null) {
       Uri uri = Uri.parse(url);
       this.mediaPlayer = MediaPlayer.create(getCurrentActivity(), uri);
+      if (this.mediaPlayer == null) return
       this.mediaPlayer.setOnCompletionListener(
         new OnCompletionListener() {
           @Override
